src/main/java/ch/njol/skript/Skript.java

```diff
@@ -1274,120 +1279,137 @@ public static SkriptAddon getAddonInstance() {
 
 	// ================ CONDITIONS & EFFECTS & SECTIONS ================
 
-	private static final Collection<SyntaxElementInfo<? extends Condition>> conditions = new ArrayList<>(50);
-	private static final Collection<SyntaxElementInfo<? extends Effect>> effects = new ArrayList<>(50);
-	private static final Collection<SyntaxElementInfo<? extends Statement>> statements = new ArrayList<>(100);
-	private static final Collection<SyntaxElementInfo<? extends Section>> sections = new ArrayList<>(50);
-
 	/**
 	 * registers a {@link Condition}.
 	 * 
-	 * @param condition The condition's class
+	 * @param conditionClass The condition's class
 	 * @param patterns Skript patterns to match this condition
 	 */
-	public static <E extends Condition> void registerCondition(final Class<E> condition, final String... patterns) throws IllegalArgumentException {
-		checkAcceptRegistrations();
-		String originClassPath = Thread.currentThread().getStackTrace()[2].getClassName();
-		final SyntaxElementInfo<E> info = new SyntaxElementInfo<>(patterns, condition, originClassPath);
-		conditions.add(info);
-		statements.add(info);
+	public static <E extends Condition> void registerCondition(Class<E> conditionClass, String... patterns) throws IllegalArgumentException {
+		String originClass = Thread.currentThread().getStackTrace()[2].getClassName();
+		SyntaxInfo<E> info = SyntaxInfo.of(BukkitOrigin.of(originClass),
+				conditionClass, ImmutableList.copyOf(patterns));
```

Comment:
I might be wrong but isn't ImmutableList more expensive than something like `Collections.unmodifiableList`? I think in this case the later would suffice?

Reply:
It's for converting String[] into a list, might just do with new ArrayList instead.